### PR TITLE
libxcrypt@4.4.36.bcr.1

### PR DESCRIPTION
**Pull request by malt3**
Adds `config.h` for macOS (Intel and Apple Silicon).
This was tested on both Intel and Apple Silicon Macs.

This will allow building `@openssh//:sshd` on macOS after this and #3149 have been merged.

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @mattyclarkson, modules you maintain (libxcrypt) have been updated in this PR. Please review the changes.

**Comment by mattyclarkson** [comment]
LGTM, thanks!

**Review by meteorcloudy — DISMISSED** [review]

**Comment by malt3** [comment]
I backported [this commit](https://github.com/besser82/libxcrypt/commit/9a8ba6ed1c60b77f05c8163e0b6a5f23aeecb993) to avoid `#include <crypt.h>` in a test.

@meteorcloudy is there a way I can easily run `bazel test @libxcrypt//...` with the dev dependencies of libxcrypt (`rules_python`) locally?
If I add libxcrypt as a `bazel_dep` in my own MODULE, I am unable to run the tests due to missing dev dependencies.

**Comment by meteorcloudy** [comment]
No, the only way to fetch dev dependency is to run the module as the root module, which is doable using `bcr_test_module` and set `module_path` to `.`. See https://github.com/bazelbuild/bazel-central-registry/blob/main/docs/README.md#test-module

File: modules/libxcrypt/4.4.36.bcr.1/presubmit.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+bcr_test_module:
+  module_path: ""
+  matrix:
+    bazel:
+      - 7.x
+      - 8.0.0rc1
```
Comment (meteorcloudy) [inline modules/libxcrypt/4.4.36.bcr.1/presubmit.yml:6]: nit: the latest rc is 8.0.0rc2

**Review by meteorcloudy — DISMISSED** [review]

**Review by meteorcloudy — APPROVED** [review]

**Pull request merged**